cli/cmd/install.go

```diff
@@ -23,117 +23,148 @@ import (
+		err := getRepo("edgeless", "https://helm.edgeless.systems/stable", options.settings)
 		if err != nil {
 			return err
 		}
-		path, err = installer.ChartPathOptions.LocateChart("edgeless/marblerun-coordinator", settings)
+		options.chartPath, err = installer.ChartPathOptions.LocateChart("edgeless/marblerun-coordinator", options.settings)
 		if err != nil {
 			return err
 		}
 	}
-	chart, err := loader.Load(path)
+	chart, err := loader.Load(options.chartPath)
 	if err != nil {
 		return err
 	}
 
-	if err := chartutil.ValidateAgainstSchema(chart, vals); err != nil {
+	resourceKey, err := getResourceKey(options.defaultKey)
+	if err != nil {
 		return err
 	}
 
-	if _, err := installer.Run(chart, vals); err != nil {
-		return err
+	// set overwrite values
+	finalValues := map[string]interface{}{}
+	var stringValues []string
+
+	stringValues = append(stringValues, fmt.Sprintf("coordinator.meshServerPort=%d", options.meshPort))
+	stringValues = append(stringValues, fmt.Sprintf("coordinator.clientServerPort=%d", options.clientPort))
+
+	if options.simulation {
+		// simulation mode, disable tolerations and resources, set simulation to 1
+		stringValues = append(stringValues,
+			fmt.Sprintf("tolerations=%s", "null"),
+			fmt.Sprintf("coordinator.simulation=%d", 1),
+			fmt.Sprintf("coordinator.resources.limits=%s", "null"),
+			fmt.Sprintf("coordinator.hostname=%s", options.hostname),
+		)
+	} else {
+		stringValues = append(stringValues,
+			fmt.Sprintf("tolerations[0].key=%s", resourceKey),
+			fmt.Sprintf("tolerations[0].operator=Exists"),
+			fmt.Sprintf("tolerations[0].effect=NoSchedule"),
+			fmt.Sprintf("coordinator.hostname=%s", options.hostname),
+		)
+	}
+
+	for _, val := range stringValues {
+		if err := strvals.ParseInto(val, finalValues); err != nil {
+			return err
+		}
+	}
+
+	// strvals cant parse keys which include dots, e.g. setting as a resource limit key "sgx.intel.com/epc" will lead to errors
+	// With this we directly set the needed values as a map[string]interface
+	if !options.simulation {
+		// Since we want to potentially change a key in the "limits" map we have to remove keys that are different from the one we want to insert
+		// If we just set a new key the resulting deployment will potentially have 2 resource limits for different SGX definitions resulting in a pod that will never start
+		if presetLimits, ok := chart.Values["coordinator"].(map[string]interface{})["resources"].(map[string]interface{})["limits"].(map[string]interface{}); ok {
+			finalValues["coordinator"].(map[string]interface{})["resources"] = map[string]interface{}{
+				"limits": map[string]interface{}{},
+			}
+			for oldResourceKey := range presetLimits {
+				// Make sure the key we delete is not a common kubernetes resource type
+				if oldResourceKey != resourceKey && !(strings.Contains("cpu memory", oldResourceKey) || strings.Contains(oldResourceKey, "hugepages-")) {
```

Comment:
This is dangerous, imagine someone using other custom resources.
The scenario you're trying to prevent is someone setting a `kubernetes.azure.com/sgx_epc_mem_in_MiB` limit in their k8s file but installing marblerun with the default resourceKey?
I'd suggest we disable the injection of limits and tolerations by default and if someone enables them, we leave it to them to set the correct resource key.

Reply:
Do we instead just install the coordinator using the values provided by the helm chart?
Another approach could be checking the resource keys of the helm chart against a list of known sgx keys (azure, intel etc.) and then removing those, provided they are different from what we want to set as the new value. This should prevent accidentally removing custom resource requests a user might have set.

The change i would make is:
* check if the `resource-key` flag was set by a user or left empty
* if it is empty, check the cluster for sgx resources and return the key (`sgx.intel.com/epc`), if we cant find any return a default value
* if it was set, return that value as the key
* set the returned value as toleration for the coordinator
* if the coordinator has any sgx resource requests different from the returned key remove them

Then again we could just start the coorindator likei it is defined in the helm chart and only if a user sets the `resource-key` flag we change tolerations/resource limits

Reply:
The downside of your suggestion would be that we need to update this code as well if we want to add a new plugin?
I don't believe this will happen very often with the official Intel plugin, so your suggestion would be fine with me:-)